### DISCUSSION TITLE
Customizing and Optimizing GitHub Dashboard Feeds

### DISCUSSION THREAD

**Discussion by bunkid** [community/community#148223] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I'm looking to make my GitHub Dashboard Feed more efficient for daily use. I want to prioritize updates from specific repositories or teams while filtering out less relevant notifications.

Are there built-in options for customization?
How can I use the Feeds API to create a personalized feed or integrate it into another dashboard application?
Any tips, tools, or examples would be greatly appreciated!

**Comment by levidang306** [comment, marked as answer]
Customizing and Optimizing GitHub Dashboard Feeds
1. Customizing Your GitHub Dashboard Feed:

GitHub offers limited customization options for the Dashboard Feed directly. However, you can prioritize the updates you see through these methods:

Star Repositories: By starring repositories, you'll make sure that updates from those repos are highlighted in your Dashboard Feed. This ensures that key projects are always at the top.
Follow Organizations and Users: Following specific users or organizations can help filter your feed to show relevant updates, as you'll receive notifications for their activity.
Notifications Settings: Adjust your notification settings to get notified about certain activities like issues, pull requests, releases, etc. This helps limit notifications to what's important to you.
Saved Searches: Although GitHub doesn't allow direct customization of the Dashboard Feed, you can create saved searches for issues or pull requests in specific repositories that you follow.
2. Using the Feeds API for Personalization:

GitHub's Feeds API provides endpoints for accessing feed data, such as events in repositories, organizations, or user activities. To use this for creating personalized feeds or integrating into external applications:

GitHub Activity Feed: The Feeds API allows you to pull activity data, which you can use to create custom views or dashboards outside of GitHub. You can query the events of specific users or repositories.

Example API request for events related to a user:

bash
Copy code
GET https://api.github.com/users/username/events
This will return a list of public events for that user.

Custom Dashboards: You can build a custom application or integrate this data into an existing dashboard (like in Slack or other project management tools) to track your repositories and teams. Using the API, you can filter by event types, such as push, pull_request, or issue_comment, and focus only on the activities that matter most.

Webhooks: For real-time updates, you can use GitHub Webhooks to get notified when something happens in your repositories. This can feed data directly into your application, so you always have up-to-date information.

3. Additional Tips:

Use GitHub Actions to automate processes and reduce manual intervention in monitoring feeds.
Explore third-party integrations such as Zapier or IFTTT, which allow you to pull GitHub updates into other apps or filter them based on your custom preferences.

**Reply by bunkid** [reply]
Thank you.
